spec/requests/maintenance_mode_spec.rb

```diff
@@ -0,0 +1,40 @@
+require "rails_helper"
+
+RSpec.describe "Maintenance Mode", type: :request do
+  context "when MAINTENANCE_MODE is set" do
+    before do
+      allow(Rails.application.config.x).to receive(:maintenance_mode) { true }
```

Comment:
A common pattern for a temporary overriding a configuration value without stubbing magic is something like this:
```suggestion
  @original_maintenance_mode_value = Rails.application.config.x.maintenance_mode
  Rails.application.config.x.maintenance_mode = true
```
and then you can reset in the `after` call:
```
after do
  Rails.application.config.x.maintenance_mode = @original_maintenance_mode_value
end
```